lib/spack/spack/cmd/install.py

```diff
@@ -88,4 +91,6 @@ def install(parser, args):
                 verbose=args.verbose,
                 fake=args.fake,
                 dirty=args.dirty,
-                explicit=True)
+                explicit=True,
+                stop_at=args.stop_at
```

Comment:
> You're adding a new mechanic that does something identical, but with a completely different syntax.

Here I don't share your point of view. In this PR I am adding phases to the `install` method, specializing  for different build-systems : thus I permit to stop at each phase in the `install` command. 

Besides being possibly of help for debugging a package, this option served the purpose of showing that I don't need everything to be named the same for it to work. The `fetch`, `stage`, `patch` actions are instead preparatory actions for `install` that are managed by `spack`, not by the build-system. 

I would rather make them preparatory `InstallPhase`s shared by all the sub-classes (and eliminate or deprecate a few commands in the meanwhile), rather than the opposite. And of course I would do so in another PR...

> We would have to name the methods similarly for each package subclass.

Did I already mention that I am REALLY against that ? :smile: 

> What will happen when I try to run/stop at a stage that doesn't exist?

```
$ spack install --stop-at configure astyle
==> Installing astyle
==> Error: 'configure' is not among the allowed phases for package astyle
```
My intention would be to modify `spack info` to show information about the allowed phases. Didn't do that yet...